### PR TITLE
test: add stubbing for datasets creation & update

### PR DESCRIPTION
**Pull request by csm-thu**

### CODE REVIEW

File: cypress/commons/utils/apiUtils.js [hunk 1]
```diff
@@ -286,6 +294,71 @@ const interceptGetDatasets = () => {
...
+//   - fileName: name (or path in Storage) of the uploaded file
+//   - fileContent: content of the uploaded file
+const interceptUploadWorkspaceFile = (fileName, fileContent) => {
+  const alias = forgeAlias('reqUploadWorkspaceFile');
+  cy.intercept({ method: 'POST', url: API_REGEX.FILE_UPLOAD, times: 1 }, (req) => {
+    if (stub.isEnabledFor('CREATE_DATASET')) {
+      stub.addWorkspaceFile(fileName, fileContent);
+      req.reply(fileName);
+    } else if (stub.isEnabledFor('GET_DATASETS')) {
+      req.continue((res) => stub.addWorkspaceFile(fileName, fileContent));
```
Comment (esasova) [inline cypress/commons/utils/apiUtils.js:356]: `res` argument isn't used, it is mandatory for continue function? :thinking:

**Review by esasova — APPROVED** [review]
LGTM :+1:

File: cypress/commons/utils/apiUtils.js (hunk 1 above)
Comment (csm-thu) [reply cypress/commons/utils/apiUtils.js:356]: We can probably remove it from the arguments :+1:

File: cypress/commons/actions/generic/ScenarioParameters.js [hunk 2]
```diff
@@ -125,14 +125,30 @@ function launch(options) {
...
+//    true if the option 'datasetsEvents' is set.
 //  - updateOptions: options to provide to the interception of the "scenario update" query (default: undefined)
-function save(options) {
+//  - datasetsEvents: list of objects describing dataset-related queries to intercept; objects have this structure;
+//    - id (optional): id of the dataset to create
+//    - onDatasetCreation (optional): validation function to run on the dataset creation request
+//    - onDatasetUpdate (optional): validation function to run on the dataset update request
+function save(options = {}) {
+  const aliases = [];
+  // Events array is reversed to make tests easier to wrtie and still match the order of cypress interceptionss
```
[the remark this answers is not in the document]
Comment (csm-thu) [reply cypress/commons/actions/generic/ScenarioParameters.js:137]: Too many typos for one line :sweat_smile:

[the remark this answers is not in the document]
Comment (csm-thu) [reply cypress/commons/actions/generic/ScenarioParameters.js]: Fixed :+1:

**Review by nborde-CSM — APPROVED** [review]
LGTM :+1:

**Pull request merged**